frontend/src/model/beta/overview/TemplatePage.tsx

```diff
@@ -12,9 +14,11 @@ type TemplatePageProps = {
 export default function TemplatePage({ model }: TemplatePageProps) {
   const theme = useTheme()
   const router = useRouter()
+  const [loading, setLoading] = useState(false)
 
   function handleCreateFromScratchClick() {
     router.push(`/beta/model/${model.id}/schema`)
+    setLoading(true)
```

Comment:
Quick note that this line is pointless. `router.push` will change the page the user is on, so will render a different set of components. As this component is no longer being rendered (and therefore the button is no longer being rendered) the loading state won't be shown to the user.